### PR TITLE
Set partial-data-available on created company

### PR DESCRIPTION
**Pull request by gzcomphouse**
Certain company types have a value set in the `partial-data-available` field. This change adds the correct value to these certain companies on creation.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/testdata/model/entity/CompanyProfile.java [hunk 1]
```diff
@@ -331,4 +342,15 @@ public Boolean getIsCommunityInterestCompany() {
     public void setIsCommunityInterestCompany(Boolean isCommunityInterestCompany) {
         this.isCommunityInterestCompany = isCommunityInterestCompany;
     }
+
+    public String getPartialDataAvailable() {return this.partialDataAvailable; }
+
+    public void setPartialDataAvailable(String companyType) {
```
Comment (lyadavalli) [inline src/main/java/uk/gov/companieshouse/api/testdata/model/entity/CompanyProfile.java:348]: can we move this to companyprofileimpl class and also you can create a private method for setting these values

File: src/main/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyProfileServiceImpl.java [hunk 2]
```diff
@@ -69,6 +69,7 @@ public CompanyProfile create(CompanySpec spec) {
         profile.setCompanyNumber(companyNumber);
         profile.setDateOfCreation(dateOneYearAgo);
         profile.setType(Objects.requireNonNullElse(companyType, "ltd"));
+        profile.setPartialDataAvailable(profile.getType());
```
Comment (lyadavalli) [inline src/main/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyProfileServiceImpl.java:72]: no need to set empty value if partial data is not available.. please add condition to check this to avoid empty value in database

File: src/test/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyProfileServiceImplTest.java [hunk 3]
```diff
@@ -316,4 +318,46 @@ void createCompanyWithSuperSecurePscsNull() {
...
+        CompanyProfile returnedProfile = this.companyProfileService.create(spec);
+        assertEquals(savedProfile, returnedProfile);
+        ArgumentCaptor<CompanyProfile> companyProfileCaptor = ArgumentCaptor.forClass(CompanyProfile.class);
+        verify(repository).save(companyProfileCaptor.capture());
+
+        CompanyProfile profile = companyProfileCaptor.getValue();
+        assertEquals(COMPANY_TYPE_INDUSTRIAL_AND_PROVIDENT_SOCIETY, profile.getType());
+        assertEquals("full-data-available-from-financial-conduct-authority-mutuals-public-register", profile.getPartialDataAvailable());
+    }
 }
```
Comment (lyadavalli) [inline src/test/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyProfileServiceImplTest.java:363]: please add test for if partial data is null

**Review by lyadavalli — CHANGES_REQUESTED** [review]

**Review by lyadavalli — CHANGES_REQUESTED** [review]
Please update the ReadMe file

File: src/main/java/uk/gov/companieshouse/api/testdata/model/entity/CompanyProfile.java (hunk 1 above)
Comment (gzcomphouse) [reply src/main/java/uk/gov/companieshouse/api/testdata/model/entity/CompanyProfile.java:348]: Have moved to this class

File: src/main/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyProfileServiceImpl.java (hunk 2 above)
Comment (gzcomphouse) [reply src/main/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyProfileServiceImpl.java:72]: Resolved

File: src/test/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyProfileServiceImplTest.java (hunk 3 above)
Comment (gzcomphouse) [reply src/test/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyProfileServiceImplTest.java:363]: Added

File: src/main/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyProfileServiceImpl.java [hunk 4]
```diff
@@ -101,6 +109,22 @@ public CompanyProfile create(CompanySpec spec) {
...
+        if (companyJurisdiction.equals("northern-ireland")) {
+            partialDataOptions.put("industrial-and-provident-society",
+                    "full-data-available-from-department-of-the-economy");
+        } else {
+            partialDataOptions.put("industrial-and-provident-society",
+                    "full-data-available-from-financial-conduct-authority-mutuals-public-register");
+        }
+        return partialDataOptions;
+    }
+
```
Comment (pjha-ch) [inline src/main/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyProfileServiceImpl.java:127]: All hard-coded strings should be extracted into constants `private static final String`

Comment (pjha-ch) [reply src/main/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyProfileServiceImpl.java:127]: Utilise the existing enum for Jurisdiction 

`src/main/java/uk/gov/companieshouse/api/testdata/model/rest/Jurisdiction.java`

File: src/test/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyProfileServiceImplTest.java [hunk 5]
```diff
@@ -316,4 +318,88 @@ void createCompanyWithSuperSecurePscsNull() {
...
+
+        CompanyProfile returnedProfile = this.companyProfileService.create(spec);
+        assertEquals(savedProfile, returnedProfile);
+        ArgumentCaptor<CompanyProfile> companyProfileCaptor = ArgumentCaptor.forClass(CompanyProfile.class);
+        verify(repository).save(companyProfileCaptor.capture());
+
+        CompanyProfile profile = companyProfileCaptor.getValue();
+        assertEquals(COMPANY_TYPE_LTD, profile.getType());
+        assertNull(profile.getPartialDataAvailable());
+    }
```
Comment (pjha-ch) [inline src/test/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyProfileServiceImplTest.java:404]: There's a lot of duplicate code in the added tests. Consider extracting common steps into a separate method to improve maintainability.

**Review by pjha-ch — CHANGES_REQUESTED** [review]
Please have a look at my review comments.
